src/reports/data/node.cpp

```diff
@@ -2,23 +2,18 @@
 
 #include "node.hpp"
 
-Node::Node(uint64_t gid, uint64_t vgid): m_gid(gid), m_vgid(vgid) {
-}
-
 void Node::add_element(double* element_value, uint32_t element_id) {
     m_elements.push_back(element_value);
     m_element_ids.push_back(element_id);
 }
 
-int Node::fill_data(double* data, bool node_to_be_recorded) {
+size_t Node::fill_data(double* data) {
     // Copy data from node data structures to buffer
-    int reporting_total = 0;
-    if(node_to_be_recorded) {
-        for (auto &elem: m_elements) {
-            data[reporting_total++] = *elem;
-        }
+    for (auto &elem: m_elements) {
+        0[data++] = *elem;
```

Comment:
You can refuse this syntax if you want, I will modify it so.

Reply:
I prefer the STL way of life to replace the entire loop. Less code, less bugs

```cpp
std::transform(m_elements.begin(), m_elements.end(), data, [](auto elem) { return *elem; });
```

Reply:
It also applies to the `Node::refresh_pointers` member method.
```cpp
void Node::refresh_pointers(refresh_function_t refresh_function) {
    std::transform(m_elements.begin(), m_elements.end(), m_elements.begin(), refresh_function);
}
```

Reply:
I find this syntax really confusing.
I think it would be better *(data++) = *elem, as you were discussing in Slack with Omar.

Reply:
I didnt read the tristan comment before my previous comment, but I think is a good idea to use stl transform in simple those simple loops

Reply:
Ok @tristan0x

Reply:
I so much want ranges from c++20!